src/schedule/zephyr_ll.c

```diff
@@ -0,0 +1,455 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Guennadi Liakhovetski <guennadi.liakhovetski@linux.intel.com>
+
+#include <sof/list.h>
+#include <sof/spinlock.h>
+#include <sof/audio/component.h>
+#include <sof/drivers/interrupt.h>
+#include <sof/lib/notifier.h>
+#include <sof/schedule/ll_schedule_domain.h>
+#include <sof/schedule/schedule.h>
+#include <sof/schedule/task.h>
+
+#include <kernel.h>
+
+/* per-scheduler data */
+struct zephyr_ll {
+	struct list_item tasks;			/* list of ll tasks */
+	unsigned int n_tasks;			/* task counter */
+	spinlock_t lock;			/* protects the list of tasks and the counter */
+	struct ll_schedule_domain *ll_domain;	/* scheduling domain */
+};
+
+/* per-task scheduler data: so far actually unused */
```

Comment:
Should this be in the next PR ?

Reply:
Actually this data is already used, I'll update the comment